apps/emqx/src/emqx_frame.erl

```diff
@@ -439,6 +470,13 @@ parse_topic_filters(unsubscribe, Bin) ->
 parse_reason_codes(Bin) ->
     [Code || <<Code>> <= Bin].
 
+parse_utf8_pair(Bin)
+  when 4 > byte_size(Bin) ->
+    ?PARSE_ERR(utf8_pair_length_error);
+parse_utf8_pair(Bin = <<Len1:16/big, _Key:Len1/binary,
+                        Len2:16/big, _Val:Len2/binary, _Rest/binary>>)
+  when Len1 + Len2 > byte_size(Bin) ->
```

Comment:
this can never happen as the binary match expression already guaranteed `Len1 + Len2 + 4 =< byte_size(Bin)`

Reply:
maybe add this after the normal clause:
```
parse_utf8_pair(<<LenK:16/big, _:Len/binary, LenV:16/big, Rest/binary>>) when LenV > byte_size(Rest) ->
    ?PARSE_ERR(#{hint => malformed_user_property_value,
                 parsed_key_length => LenK, %% do not log parsed key because it's likely malformed
                 parsed_value_length => LenV,
                 remaining_bytes => byte_size(Rest)});
parse_utf8_pair(<<Len:16/big, Rest/binary>>) when Len > byte_size(Rest) ->
    ?PARSE_ERR(#{hint => malformed_user_property_key,
                 parsed_length => Len,
                 remaining_bytes => byte_size(Rest)});
parse_utf8_pair(Bytes) when byte_size(Bytes) < 4 ->
    ?PARSE_ERR(#{hint => malformed_user_property,
                 total_bytes => byte_size(Rest)});

```